packages/typeahead/src/react/index.tsx

```diff
@@ -84,215 +81,224 @@ const defaultFilterFunc = (
+      React.ReactText | undefined
+    >(value || '')
+    const [inputItems, setInputItems] = React.useState(options)
+    const {
+      closeMenu,
+      getToggleButtonProps,
+      getComboboxProps,
+      getInputProps,
+      getItemProps,
+      getLabelProps,
+      getMenuProps,
+      highlightedIndex,
+      isOpen,
+      openMenu,
+      selectedItem: activeItem
+    } = useCombobox({
+      defaultHighlightedIndex: 0,
+      items: inputItems,
+      stateReducer(state, actionAndChanges) {
+        const { type, changes } = actionAndChanges
+        // this prevents the menu from being closed when the user selects an item with 'Enter' or mouse
+        switch (type) {
+          case useCombobox.stateChangeTypes.InputBlur:
+          case useCombobox.stateChangeTypes.InputKeyDownEnter:
+          case useCombobox.stateChangeTypes.ItemClick: {
+            const { selectedItem, inputValue } = changes
+            return {
+              ...changes,
+              inputValue: selectedItem?.label
+                ? `${selectedItem?.label}`
+                : inputValue
+            }
           }
+          default:
+            return changes // otherwise business as usual.
         }
-        default:
-          return changes // otherwise business as usual.
-      }
-    },
-    onInputValueChange: ({ type, inputValue, selectedItem }) => {
-      const bothTypeAutoComplete =
+      },
+      onInputValueChange: ({ type, inputValue, selectedItem }) => {
+        const bothTypeAutoComplete =
+          type === '__input_blur__' &&
+          !selectedItem &&
+          !inputValue &&
+          ariaAutoComplete === 'both'
+        setSearchTerm(bothTypeAutoComplete ? searchTerm : inputValue)
         type === '__input_blur__' &&
-        !selectedItem &&
-        !inputValue &&
-        ariaAutoComplete === 'both'
-      setSearchTerm(bothTypeAutoComplete ? searchTerm : inputValue)
-      type === '__input_blur__' &&
-        onChange &&
-        onChange(
-          null,
-          bothTypeAutoComplete
-            ? { label: searchTerm as string }
-            : selectedItem || undefined
-        )
-      setInputItems(filterFunction(options, inputValue))
-    }
-  })
-  React.useEffect(() => {
-    onChange && onChange(null, activeItem || undefined)
-  }, [activeItem])
-  const { value: inputValue, ...inputProps } = getInputProps({
-    onKeyDown: (evt: React.KeyboardEvent<HTMLInputElement>) => {
-      if (!canUseDOM()) return
+          onChange &&
+          onChange(
+            null,
+            bothTypeAutoComplete
+              ? { label: searchTerm as string }
+              : selectedItem || undefined
+          )
+        setInputItems(filterFunction(options, inputValue))
+      }
+    })
+    React.useEffect(() => {
+      onChange && onChange(null, activeItem || undefined)
+    }, [activeItem, onChange])
+    const { value: inputValue, ...inputProps } = getInputProps({
+      ref: forwardedRef,
```

Comment:
This is the key line change in the body of the function.

The rest is formatting from the identation change because of the forwardRef wrapping.